app/services/booklet_pdf_service.rb

```diff
@@ -224,6 +224,10 @@ def participant_confirmed_count
     "#{num_participants} confirmed / #{proposal.proposal_type&.participant} maximum participants \\\\ \n"
   end
 
+  def first_subject_proposal
+    @proposals = @subjects_with_proposals.first[1].min_by(&:code).id
+  end
+
```

Comment:
What is this doing?

Reply:
Getting first proposal after sorting on subject and then writing to File, after first proposal we append other proposals in file.

Reply:
So it sets `@proposals` to the id of the first proposal in the collection, to be used ... where? 

Is `.min_by(&:code)` just to ensure that it is a proposal that has a code? Is it possible to print a book of Draft proposals?

Reply:
It's used in `check_no_latex` method.

Reply:
I would guess that a method named `check_no_latex` would be checking whether the "no latex" option is used. However, it appears to check whether the id of the first proposal from the collection of proposals sorted by subject is equal to `@current_proposal.id` (where is @current_proposal set?), and if so, writes a file instead of appends to it. A better name for this method would be `write_or_append(proposal, text)`

Where does `@current_proposal` come from?

Reply:
Oh I see now, the method `read_file` checks for no_latex.